libafl/src/feedbacks/map.rs

```diff
@@ -393,11 +371,14 @@ where
         let initial = observer.initial();
 
         let map_state = state
-            .feedback_states_mut()
-            .match_name_mut::<MapFeedbackState<T>>(&self.name)
+            .named_metadata_mut()
+            .get_mut::<MapFeedbackMetadata<T>>(&self.name)
             .unwrap();
+        if map_state.history_map.len() < observer.len() {
+            map_state.history_map.resize(observer.len(), T::default());
+        }
 
-        assert!(size <= map_state.history_map.len(), "The size of the associated map observer cannot exceed the size of the history map of the feedback. If you are running multiple instances of slightly different fuzzers (e.g. one with ASan and another without) synchronized using LLMP please check the `configuration` field of the LLMP manager.");
+        // assert!(size <= map_state.history_map.len(), "The size of the associated map observer cannot exceed the size of the history map of the feedback. If you are running multiple instances of slightly different fuzzers (e.g. one with ASan and another without) synchronized using LLMP please check the `configuration` field of the LLMP manager.");
 
         assert!(size <= observer.len());
```

Comment:
it's unrelated to PR change.
but I think observer.len() >= size (observer.usable_count()) is always true(?)

Reply:
only for stdmapobserver